tests/bsim/bluetooth/host/misc/conn_stress/peripheral & central: memcpy out of bounds

**Describe the bug**
For both tests/bsim/bluetooth/host/misc/conn_stress/peripheral & central tests,
the test code is memcpy'ing outside of the input buffer.

* tests/bsim/bluetooth/host/misc/conn_stress/peripheral
* tests/bsim/bluetooth/host/misc/conn_stress/central

**To Reproduce**
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -GNinja -DBOARD=nrf52_bsim ../tests/bsim/bluetooth/host/misc/conn_stress/central && ninja 
3. See error

AND

1. mkdir build; cd build
2. cmake -GNinja -DBOARD=nrf52_bsim ../tests/bsim/bluetooth/host/misc/conn_stress/peripheral && ninja 
3. See error


**Expected behavior**
No build warnings and no reads out of bounds

**Impact**
Likely random crashes on the tests depending on host compiler, and unrelated code changes.

**Logs and console output**
For central:
```
[53/196] Building C object CMakeFiles/app.dir/src/main.c.obj
tests/bsim/bluetooth/host/misc/conn_stress/central/src/main.c: In function ‘discover_func’:
tests/bsim/bluetooth/host/misc/conn_stress/central/src/main.c:253:17: warning: ‘memcpy’ reading 17 bytes from a region of size 4 [-Wstringop-overread]
  253 |                 memcpy(&conn_info_ref->uuid, BT_UUID_GATT_CCC, sizeof(conn_info_ref->uuid));
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from include/zephyr/bluetooth/gatt.h:28,
                 from tests/bsim/bluetooth/host/misc/conn_stress/central/src/main.c:15:
include/zephyr/bluetooth/uuid.h:113:64: note: source object ‘<U76c0>’ of size 4
  113 |         ((const struct bt_uuid *) ((const struct bt_uuid_16[]) {BT_UUID_INIT_16(value)}))
      |                                                                ^
include/zephyr/bluetooth/uuid.h:887:9: note: in expansion of macro ‘BT_UUID_DECLARE_16’
  887 |         BT_UUID_DECLARE_16(BT_UUID_GATT_CCC_VAL)
      |         ^~~~~~~~~~~~~~~~~~
tests/bsim/bluetooth/host/misc/conn_stress/central/src/main.c:253:46: note: in expansion of macro ‘BT_UUID_GATT_CCC’
  253 |                 memcpy(&conn_info_ref->uuid, BT_UUID_GATT_CCC, sizeof(conn_info_ref->uuid));
      |                                              ^~~~~~~~~~~~~~~~
```

For peripheral
```
[82/195] Building C object CMakeFiles/app.dir/src/main.c.obj
tests/bsim/bluetooth/host/misc/conn_stress/peripheral/src/main.c: In function ‘discover_func’:
tests/bsim/bluetooth/host/misc/conn_stress/peripheral/src/main.c:280:17: warning: ‘memcpy’ reading 17 bytes from a region of size 4 [-Wstringop-overread]
  280 |                 memcpy(&uuid, BT_UUID_GATT_CCC, sizeof(uuid));
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from include/zephyr/bluetooth/gatt.h:28,
                 from tests/bsim/bluetooth/host/misc/conn_stress/peripheral/src/main.c:15:
include/zephyr/bluetooth/uuid.h:113:64: note: source object ‘<U46c0>’ of size 4
  113 |         ((const struct bt_uuid *) ((const struct bt_uuid_16[]) {BT_UUID_INIT_16(value)}))
      |                                                                ^
include/zephyr/bluetooth/uuid.h:887:9: note: in expansion of macro ‘BT_UUID_DECLARE_16’
  887 |         BT_UUID_DECLARE_16(BT_UUID_GATT_CCC_VAL)
      |         ^~~~~~~~~~~~~~~~~~
tests/bsim/bluetooth/host/misc/conn_stress/peripheral/src/main.c:280:31: note: in expansion of macro ‘BT_UUID_GATT_CCC’
  280 |                 memcpy(&uuid, BT_UUID_GATT_CCC, sizeof(uuid));
      |                               ^~~~~~~~~~~~~~~~
```

**Environment (please complete the following information):**
 - OS: Linux (Ubuntu 22.04)
 - Toolchain (e.g Zephyr SDK, ...): gcc 12
 - Commit SHA or Version used: Todays main: 6e678e3baee0c57fcf9a96e47b79f8b9a9e6eeaf

--- a/tests/bsim/bluetooth/host/misc/conn_stress/central/src/main.c
+++ b/tests/bsim/bluetooth/host/misc/conn_stress/central/src/main.c
@@ -96,6 +96,8 @@ static struct conn_info conn_infos[CONFIG_BT_MAX_CONN] = {0};
 static uint32_t conn_interval_max, notification_size;
 static uint8_t vnd_value[CHARACTERISTIC_DATA_MAX_LEN];
 
+static const struct bt_uuid_16 ccc_uuid = BT_UUID_INIT_16(BT_UUID_GATT_CCC_VAL);
+
 void clear_info(struct conn_info *info)
 {
 	/* clear everything except the address + sub params + uuid (lifetime > connection) */
@@ -250,8 +252,8 @@ static uint8_t discover_func(struct bt_conn *conn, const struct bt_gatt_attr *at
 
 	} else if (conn_info_ref->discover_params.type == BT_GATT_DISCOVER_CHARACTERISTIC) {
 		LOG_DBG("Service Characteristic Found");
-		memcpy(&conn_info_ref->uuid, BT_UUID_GATT_CCC, sizeof(conn_info_ref->uuid));
-		conn_info_ref->discover_params.uuid = &conn_info_ref->uuid.uuid;
+
+		conn_info_ref->discover_params.uuid = &ccc_uuid.uuid;
 		conn_info_ref->discover_params.start_handle = attr->handle + 2;
 		conn_info_ref->discover_params.type = BT_GATT_DISCOVER_DESCRIPTOR;
 		conn_info_ref->subscribe_params.value_handle = bt_gatt_attr_value_handle(attr);

--- a/tests/bsim/bluetooth/host/misc/conn_stress/peripheral/src/main.c
+++ b/tests/bsim/bluetooth/host/misc/conn_stress/peripheral/src/main.c
@@ -94,6 +94,8 @@ static struct bt_uuid_128 uuid = BT_UUID_INIT_128(0);
 static struct bt_gatt_discover_params discover_params;
 static struct bt_gatt_subscribe_params subscribe_params;
 
+static const struct bt_uuid_16 ccc_uuid = BT_UUID_INIT_16(BT_UUID_GATT_CCC_VAL);
+
 static void vnd_ccc_cfg_changed(const struct bt_gatt_attr *attr, uint16_t value)
 {
 	central_subscription = (value == BT_GATT_CCC_NOTIFY) ? 1 : 0;
@@ -277,8 +279,8 @@ static uint8_t discover_func(struct bt_conn *conn, const struct bt_gatt_attr *at
 
 	} else if (discover_params.type == BT_GATT_DISCOVER_CHARACTERISTIC) {
 		LOG_DBG("Service Characteristic Found");
-		memcpy(&uuid, BT_UUID_GATT_CCC, sizeof(uuid));
-		params->uuid = &uuid.uuid;
+
+		params->uuid = &ccc_uuid.uuid;
 		params->start_handle = attr->handle + 2;
 		params->type = BT_GATT_DISCOVER_DESCRIPTOR;
 		subscribe_params.value_handle = bt_gatt_attr_value_handle(attr);

